frontera.md

```diff
@@ -12,25 +12,70 @@ Then execute:
 
 ```
 module load tacc-apptainer
-singularity pull -F docker://seissol/training:latest
-singularity build -f my-training.sif singularity.def
-singularity run my-training.sif
+apptainer pull -F docker://seissol/training:latest
+apptainer build -f my-training.sif singularity.def
+apptainer run my-training.sif
+```
+
+You can also use the automatically generated container after pulling the docker container 
+
+```
+module load tacc-apptainer
+apptainer pull -F docker://seissol/training:latest
+apptainer run training_latest.sif
 ```
 
 You can abort the jupyter lab with Ctrl-C, confirm with `y`.
 Now you should see a directory `seissol-training`.
 This folder should contain four directories for different scenarios.
 
+You can also open the jupyter notebook that runs on Frontera on your local machine with the following steps:
+
+Step 1: change `SHARED_PATH="/your/path/to/container/"` in line 75 of `job.jupyter` to the path where your sigularity container is built.
+Step 2: Run
+```
+sbatch job.jupyter
+```
+Step 3: Check the job status with
+```
+squeue -u $USER
+```
+Step 4: Once the status changes from `PD` to `R`, you will find the job output in a generated file `jupyter.out`.
+Step 5: Check the last few lines with
+```
+tail jupyter.out
+```
+You will get something like:
+```
+TACC: got login node jupyter port 60320
+TACC: created reverse ports on Frontera logins
+TACC: Your jupyter notebook server is now running at https://frontera.tacc.utexas.edu:60320/?token=2e0fade1f8b1ce00b303a7e97dd962c5cd10c17f03a245e8c761ca7e1d5e1597
+```
+Step 6: Paste the link to your local browser, you will have access to the Frontera environment on your local machine.
+```
+https://frontera.tacc.utexas.edu:60320/?token=2e0fade1f8b1ce00b303a7e97dd962c5cd10c17f03a245e8c761ca7e1d5e1597
```

Comment:
Works, but then I get:
![image](https://github.com/SeisSol/Training/assets/13536910/89c67be0-36e2-40b5-9dfd-db2509b739dc)

Reply:
also  note that `OMP_NUM_THREADS=4` could be increased
(could add a note on that)
![image](https://github.com/SeisSol/Training/assets/13536910/11c842ae-efd7-499b-affa-aac890632f58)

Reply:
![image](https://github.com/SeisSol/Training/assets/13536910/4274f916-ca5e-498a-9bd5-ae30abee1561)

Reply:
ok I found from slack that you have to use, e.g.
`!mpirun singularity run /work2/09160/ulrich/frontera/Training/my_training.sif -3 tpv13_training.geo`

Reply:
Here is my solution (and to be honest meta-llama's):
```python
# Define variables depending on the environment
is_remote = True  # Set to True if running remotely on Frontera, False otherwise
singularity_image = '/work2/09160/ulrich/frontera/Training/training_pr-40.sif'
# Define a function to run a command with an optional Singularity wrapper
def run_command(cmd, singularity=False):
    if singularity:
        return f"mpirun singularity run {singularity_image} {cmd}"
    else:
        return f"{cmd}
 ````
 and then:
 ````python
 gmsh_cmd = run_command("gmsh -3 tpv13_training.geo", is_remote)
!{gmsh_cmd}
````
 and also:
 ```python
 pumgen_cmd = run_command("pumgen -s msh2 tpv13_training.msh", is_remote)
seissol_cmd = run_command("OMP_NUM_THREADS=28 seissol parameters.par", is_remote)
```

Reply:
Yep, you are right. That was because it is not running in the container environment, but just on Frontera. So we will need to run with something like `mpirun apptainer run <container> pumgen ...`

Reply:
my option allows not duplicating the notebook.
@davschneller @sebwolf-de what do you think? How should we proceed?

Reply:
other option is 
!pumgen -s msh2 tpv13_training.msh
# on frontera with singularity
# !mpirun singularity run $singularity_image pumgen -s msh2 tpv13_training.msh